common/MobileApp.cpp

```diff
@@ -38,7 +38,10 @@ DocumentData & DocumentData::get(unsigned docId)
 void DocumentData::deallocate(unsigned docId)
 {
     assert(idToDocDataMap.find(docId) != idToDocDataMap.end());
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunused"
     DocumentData &d = get(docId);
```

Comment:
Urk, all these pragmas make for grim reading & maintainability. Can't we just delete this redundant line ? =) Ditto for other places where things are not needed (?) What are all the:

#pragma clang diagnostic ignored "-Wconditional-uninitialized"

warnings in fact ? - can we do something more elegant there; we really want fewer conditionals & pre-processing in the code if we can.

Reply:
OK. I was able to remove all the #pragma clang's as I found the build setting in Xcode that was causing this: I changed "Uninitialized Variables" from "Yes (Aggressive)" to "Yes".

I also added comments to the two cases that have odd code that generated warnings: the destructor case and the "comma in an if statement" case. Maybe they are cruft but maybe these specific cases were intentional so better to be safe and leave the code as is but without warnings.

Overall my goal was to cut down the number of warnings to a very small number. I found that several deprecation warnings were being missed because they scrolled out of view in Xcode and since Apple routinely deprecates things, I wanted to get those pushed to the top of the build warnings list.